mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -1622,7 +1623,20 @@ CollapseShapeOp::inferCollapsedType(RankedTensorType type,
     currentDim += dim;
   }
 
-  return RankedTensorType::get(newShape, type.getElementType());
+  auto encoding = type.getEncoding();
+  if (auto v = encoding.dyn_cast_or_null<VerifiableTensorEncoding>()) {
+    auto ignoreError = [&] {
+      auto emitter = mlir::emitError(UnknownLoc::get(type.getContext()));
+      emitter.abandon();
```

Comment:
I'm not sure about this part. This looks like a hack and just removing the encoding could be problematic. Can you put this (and all encoding-related changes) in a separate PR? Add @aartbik to the new PR, the sparse team heavily depends on tensor encodings, they may have an oppinion about this.

Reply:
Yeah, this touches on the sensitive topic that the "verifable tensor encoding" really can be used for several things, including sparsity. We should probably migrate to a more specialized solution in the long run, but for now, we typically do an audit of preserving the encoding where it makes sense.

Reply:
See https://github.com/llvm/llvm-project/pull/80871